jtec-core/src/main/java/edu/tum/sse/jtec/reporting/LCOVReport.java

```diff
@@ -0,0 +1,55 @@
+package edu.tum.sse.jtec.reporting;
+
+import java.util.List;
+
+public class LCOVReport {
+    private List<LCOVSection> lcovSections;
+
+    public LCOVReport(List<LCOVSection> lcovSections) {
+        this.lcovSections = lcovSections;
+    }
+
+    public List<LCOVSection> getLcovSections() {
+        return lcovSections;
+    }
+
+    public void setLcovSections(List<LCOVSection> lcovSections) {
+        this.lcovSections = lcovSections;
+    }
+
+    /**
+     * Get the LCOV section for the given source file.
+     * @param sourceFilePath The path to the source file
+     * @return The LCOV section if it exists, else null
+     */
+    public LCOVSection getLcovSection(String sourceFilePath) {
+        return lcovSections.stream()
+                .filter(l -> l.getSourceFilePath().equals(sourceFilePath))
+                .findFirst().orElse(null);
+    }
+
+    /**
+     * Add new LCOV section if none already exists for the same source file.
+     * @param lcovSection The LCOV section to be added
+     * @return true if new section was added
+     */
+    public boolean addLcovSection(LCOVSection lcovSection) {
+        if (getLcovSection(lcovSection.getSourceFilePath()) == null) {
```

Comment:
hm, perhaps let `getLcovSection` return an `Optional<LCOVSection>` if we always need to check for `null` values?

Reply:
Removed `addLcovSection`, not needed